python/src/ir.cc

```diff
@@ -1618,11 +1618,19 @@ void init_triton_ir(py::module &&m) {
 }
 
 void init_triton_env_vars(py::module &m) {
-  m.def("get_cache_invalidating_env_vars", []() -> std::map<std::string, bool> {
-    std::map<std::string, bool> ret;
-    for (const auto &envVar : CACHE_INVALIDATING_ENV_VARS) {
-      ret[envVar] = triton::tools::getBoolEnv(envVar);
-    }
-    return ret;
-  });
+  m.def("get_cache_invalidating_env_vars",
+        []() -> std::map<std::string, std::string> {
+          std::map<std::string, std::string> ret;
+          for (const auto &envVar : CACHE_INVALIDATING_ENV_VARS) {
+            auto strVal = triton::tools::getStrEnv(envVar);
+            if (strVal.empty())
+              continue;
+            auto boolV = triton::tools::isBoolEnv(envVar);
```

Comment:
nit: as we already know the string value of `envVar` here, can we avoid calling `std::getenv` again in `isBoolEnv`? maybe making it an API like `isBoolValue`?

Reply:
isBoolValue seems to be too general. Maybe isEnvValueBool? This is in GetEnv.hpp, so should be related to Env :] On the other hand, getStrEnv, getBoolEnv, and isBoolEnv look nice in naming (i.e if we choose to keep it as it is with an extra call to getenv)